### DISCUSSION TITLE
Embedding a matlibplot on a customtkinter canvas.

### DISCUSSION THREAD

**Discussion by codedreamin** [TomSchimansky/CustomTkinter#1916] [Q&A] [answered]
Is there a way to embed a matlibplot in a CTkCanvas? Base tkinter has the FigureCanvasTkAgg method but I haven't been able to locate something similar for customtkinter.

The other way I was considering doing it would be to draw the lines yourself using create_line, which then requires you to develop your own controls for formatting and such.

**Comment by ElectricCandlelight** [comment]
Would converting it to a PIL image object and displaying that in a label work?

**Reply by codedreamin** [reply]
Well I think that would work if you were only trying to place a graph on the canvas, but I think you would then lose all of the functionality of the graph.

To make it interactive would require you to keep destroying and recreating that image.

**Comment by ElectricCandlelight** [comment, marked as answer]
https://matplotlib.org/3.1.0/gallery/user_interfaces/embedding_in_tk_sgskip.html

**Reply by codedreamin** [reply]
Thanks! It might have been the examples I was looking at before but the one you linked helped me use FigureCanvasTkAgg to get it working.
